elastic/kibana
# [Response Ops][Task Manager] Returning status as `OK` immediately when not running background tasks

Resolves https://github.com/elastic/kibana-team/issues/764

## Summary

We've been seeing `taskManager plugin is now unavailable: Status check timed out after 30s` a lot in the serverless logs. After some investigation, these were all coming from the UI-only Kibana nodes. It looks like for background task nodes, TM status is emitted at the polling interval  (default `3s`) whereas for the UI nodes, TM status is emitted less frequently (every `1m`). The first emission always shows as unavailable because monitoring stats are not available, however this means for the UI nodes, TM emits `unavailable`, then doesn't emit `available` until one minute later, which exceeds the status check timeout.

This PR checks whether a node is UI only and emits `OK`, skipping the unavailable state, since UI nodes don't really depend on TM monitoring stats for availability.

## To Verify
1. Start ES locally
2. Add `node.roles: ["migrator"]` to your Kibana config and start Kibana (the migrator node must run first if we're running with split roles)
3. Update config to use `node.roles: ["ui"]` and restart Kibana. You should see Kibana start up with no errors in the logs about Task manager being unavailable.
